### ISSUE TITLE
`no-undefined-types` not working in all contexts despite docs

### ISSUE THREAD

**Issue by brettz9** [gajus/eslint-plugin-jsdoc#324]
Despite being advertised in the docs as supporting contexts "everywhere", `no-undefined-types` doesn't actually work in all contexts, as in this test case:

```js
    {
      code: `
          /**
           * @type {strnig}
           */
          var quux = {

          };
      `,
      errors: [
        {
          line: 3,
          message: 'The type \'strnig\' is undefined.'
        }
      ],
      rules: {
        'no-undef': 'error'
      }
    },
```

Setting `iterateAllJsdocs` was not alone sufficient as this caused some tests to fail.

(I've checked all other rules, and except for this one, I believe all rules that should work by default in all contexts are now working in all contexts.)
